src/EloquentDataTable.php

```diff
@@ -160,36 +161,48 @@ protected function joinEagerLoadedColumn($relation, $relationColumn)
                     break;
 
                 case $model instanceof HasOneOrMany:
-                    $table   = $model->getRelated()->getTable();
-                    $foreign = $model->getQualifiedForeignKeyName();
-                    $other   = $model->getQualifiedParentKeyName();
+                    $table     = $model->getRelated()->getTable();
+                    $foreign   = $model->getQualifiedForeignKeyName();
+                    $other     = $model->getQualifiedParentKeyName();
+                    $deletedAt = $this->checkSoftDeletesOnModel($model->getRelated());
                     break;
 
                 case $model instanceof BelongsTo:
-                    $table   = $model->getRelated()->getTable();
-                    $foreign = $model->getQualifiedForeignKey();
-                    $other   = $model->getQualifiedOwnerKeyName();
+                    $table     = $model->getRelated()->getTable();
+                    $foreign   = $model->getQualifiedForeignKey();
+                    $other     = $model->getQualifiedOwnerKeyName();
+                    $deletedAt = $this->checkSoftDeletesOnModel($model->getRelated());
                     break;
 
                 default:
                     throw new Exception('Relation ' . get_class($model) . ' is not yet supported.');
             }
-            $this->performJoin($table, $foreign, $other);
+            $this->performJoin($table, $foreign, $other, $deletedAt);
             $lastQuery = $model->getQuery();
         }
 
         return $table . '.' . $relationColumn;
     }
 
+    protected function checkSoftDeletesOnModel($model)
+    {
+        if (in_array('Illuminate\Database\Eloquent\SoftDeletes', class_uses($model))) {
+            return $model->getDeletedAtColumn();
```

Comment:
Does this return the fully qualified column name? Can check the codes yet but PR looks good. Thanks!